taf/git.py

```diff
@@ -589,8 +615,10 @@ def cleanup(self):
             self._pygit = None
 
     def clone(
-        self, no_checkout: bool = False, bare: Optional[bool] = False, **kwargs
+        self, no_checkout: bool = False, bare: Optional[bool] = None, **kwargs
     ) -> None:
+        if bare is None:
+            bare = self.is_bare_repository
```

Comment:
When cloning, the repository should not already exist, so it makes more sense to pass in `False` then `None` as a default value.